syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package protobuf;

message Coordinates {
  double lat = 1;
  double lng = 2;
}

message Stop {
  int32 id = 1;
  string name = 2;
  Coordinates coordinates = 3;
  bool is_consistent = 4;
}

message Bus {
  int32 id = 1;
  string name = 2;
  repeated int32 stop_id = 3;
  bool is_roundtrip = 4;
}

message Distance {
  uint64 key = 1;
  int32 val = 2;
}

message TransportCatalogue {
  repeated Stop stop = 1;
  repeated Bus bus = 2;
  repeated Distance distance = 3;
  RenderSettings render_settings = 4;
  TransportRouter router = 5;
}